core/src/main/java/org/opensearch/remote/metadata/client/impl/LocalClusterIndicesClient.java

```diff
@@ -383,4 +425,25 @@ public CompletionStage<SearchDataObjectResponse> searchDataObjectAsync(
     public void close() throws Exception {
         // No resources to close, OpenSearch manages the NodeClient
     }
+
+    @Override
+    public boolean isGlobalResource(String index, String id) {
```

Comment:
What is the intended use of this method?  Normally when I see `isFoo()` it's a quick return checking some variable, but this requires a client call and blocks any threads waiting for this status.

If we have a global cache why not just check if the key exists?